components/org.wso2.carbon.identity.data.publisher.authentication.audit/src/main/java/org/wso2/carbon/identity/data/publisher/authentication/audit/AuthenticationAuditLoggerUtils.java

```diff
@@ -63,24 +65,37 @@ public static AuthenticationAuditData createAuthenticationAudiDataObject(Event e
         Object userObj = params.get(FrameworkConstants.AnalyticsAttributes.USER);
         setUserStoreDomain(authenticationAuditData, userObj);
 
+        String authenticatedUser = null;
+        String username = null;
+        String tenantDomain = null;
+
         if (AuthenticationAuditLoggerConstants.AUDIT_AUTHENTICATION_STEP.equals(authType)) {
-            authenticationAuditData.setAuthenticatedUser(getUserNameForAuthenticationStep(params));
-            authenticationAuditData.setTenantDomain(getTenantDomainForAuthenticationStep(params));
+            authenticatedUser = getUserNameForAuthenticationStep(params);
+            username = authenticatedUser;
+            tenantDomain = getTenantDomainForAuthenticationStep(params);
+            authenticationAuditData.setAuthenticatedUser(authenticatedUser);
+            authenticationAuditData.setTenantDomain(tenantDomain);
             authenticationAuditData.setAuthenticatedIdps(getIdentityProviderForAuthenticationStep(context));
             authenticationAuditData.setStepNo(getStepNoForAuthenticationStep(context));
-
         } else if (AuthenticationAuditLoggerConstants.AUDIT_AUTHENTICATION.equals(authType)) {
             if (isUsernameEnabled) {
-                authenticationAuditData.setAuthenticatedUser(getAuthenticatedUserName(context, status));
+                authenticatedUser = getAuthenticatedUserName(context, status);
+                username = authenticatedUser;
+                authenticationAuditData.setAuthenticatedUser(authenticatedUser);
             } else {
-                authenticationAuditData.setAuthenticatedUser(getSubjectIdentifier(context, status));
+                authenticatedUser = getSubjectIdentifier(context, status);
+                username = MultitenantUtils.getTenantAwareUsername(authenticatedUser);
+                authenticationAuditData.setAuthenticatedUser(authenticatedUser);
             }
-            authenticationAuditData.setTenantDomain(getTenantDomainForAuthentication(context, params, status));
+            tenantDomain = getTenantDomainForAuthentication(context, params, status);
+            authenticationAuditData.setTenantDomain(tenantDomain);
             authenticationAuditData.setStepNo(getStepNoForAuthentication(context, status));
             authenticationAuditData.setAuthenticatedIdps(getIdentityProviderList(context, status));
-
         }
-
+        if (StringUtils.isNotBlank(tenantDomain) && StringUtils.isNotBlank(username)) {
+            String userId = IdentityUtil.getInitiatorId(username, tenantDomain);
```

Comment:
what happens if by any chance the username here is not a tenant aware username?

Reply:
userId will be null in that case. I am validating whether the userId is null before printing the log. If userId is null then the userId will be username.